website/docs/externaldata.md

```diff
@@ -306,3 +307,99 @@ There are several limitations when using external data with the mutating webhook
+```
+
+3. Generate a private key for your external data provider:
+
+```bash
+ openssl genrsa -out server.key 2048
+```
+
+4. Generate a certificate signing request (CSR) for your external data provider:
+
+> Replace `<service name>` and `<service namespace>` with the correct values.
+
+```bash
+openssl req -newkey rsa:2048 -nodes -keyout server.key -subj "/CN=<service name>.<service namespace>" -out server.csr
+```
+
+5. Generate a certificate for your external data provider:
+
+```bash
+openssl x509 -req -extfile <(printf "subjectAltName=DNS:<service name>.<service namespace>") -days 365 -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt
+```
+
+### How Gatekeeper trusts the external data provider (TLS)
+
+To enable one-way TLS, your external data provider should enable any TLS-related configurations for their HTTP server. For example, for Go's built-in [`HTTP server`](https://pkg.go.dev/net/http#Server) implementation, you can use [`ListenAndServeTLS`](https://pkg.go.dev/net/http#ListenAndServeTLS).
+
+In addition, the provider is also responsible for supplying the certificate authority (CA) certificate as part of the Provider spec so that Gatekeeper can verify the authenticity of the external data provider's certificate.
+
+The CA certificate must be encoded as a base64 string when defining the Provider spec. Run the following command to perform base64 encoding:
+
+```bash
+cat ca.crt | base64 | tr -d '\n'
+```
+
+With the encoded CA certificate, you can define the Provider spec as follows:
+
+```yaml
+apiVersion: externaldata.gatekeeper.sh/v1alpha1
+kind: Provider
+metadata:
+  name: my-provider
+spec:
+  url: https://<service-name>.<namespace>:<port>/<endpoint> # URL to the external data source (e.g., https://my-provider.my-namespace:8090/validate)
+  timeout: <timeout> # timeout value in seconds (e.g., 1). this is the timeout on the Provider custom resource, not the provider implementation.
+  caBundle: <encoded-ca-certificate> # base64 encoded CA certificate.
+```
+
+### How the external data provider trusts Gatekeeper (mTLS)
+
+Gatekeeper attaches its certificate as part of the HTTPS request to the external data provider. To verify the authenticity of the Gatekeeper certificate, the external data provider must have access to Gatekeeper's CA certificate. There are several ways to do this:
+
+1. Deploy your external data provider to the `gatekeeper-system` namespace. By default, [`cert-controller`](https://github.com/open-policy-agent/cert-controller) is used to generate and rotate Gatekeeper's webhook certificate. The content of the certificate is stored as a Kubernetes secret called `gatekeeper-webhook-server-cert` in the `gatekeeper-system` namespace. In your external provider deployment, you can access Gatekeeper's certificate by adding the following `volume` and `volumeMount` so that your server can trust Gatekeeper's CA certificate:
```

Comment:
```suggestion
1. Deploy your external data provider to the `gatekeeper-system` namespace. By default, [`cert-controller`](https://github.com/open-policy-agent/cert-controller) is used to generate and rotate Gatekeeper's webhook certificate. The content of the certificate is stored as a Kubernetes secret called `gatekeeper-webhook-server-cert` in the Gatekeeper namespace e.g.  `gatekeeper-system` . In your external provider deployment, you can access Gatekeeper's certificate by adding the following `volume` and `volumeMount` so that your server can trust Gatekeeper's CA certificate:
```